File: 2348_Number of Zero-Filled Subarrays.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var zeroFilledSubarray = function(nums) {
    //Approach 1
    
    var count = 0;
    for(let i = 0; i < nums.length; i++){
        if(nums[i] == 0){
            count += 1;
            for(let j = i+1; j < nums.length; j++){
                if(nums[j] == 0)
                {
                    count += 1;
                }
                else{
                    j = nums.length;
                }
            }
        }   
    }
    
    return count;

    //Approach 2
    
    let count = 0;
    let length = 0;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] === 0) {
            length++;
            count += length;
        } else length = 0;
    }
    return count;

};

File: 2348_Number of Zero-Filled Subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        length = 0
        count = 0
        for i in nums:
            if i == 0:
                length += 1
                count += length
            else:
                length = 0
        return count
# Beats 100% in leetcode               